### PR TITLE
Add harvester dependency chart

### PR DESCRIPTION
**Pull request by guangbochen**
- move dependency charts to the `dependency_chart` and use `helm dep build` by following the helm best practice.
- add harvester dependency chart of multus-cni v3.6
- bump minio to the latest version of v8.0.5
- remove the redundant symlinks

https://github.com/rancher/harvester/issues/292

### CODE REVIEW

File: deploy/charts/harvester/Chart.lock [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  version: 0.1.0
+- name: multus
+  repository: file://dependency_charts/multus
+  version: 0.1.0
+- name: minio
+  repository: https://helm.min.io
+  version: 8.0.5
+- name: longhorn
+  repository: https://charts.longhorn.io
+  version: 1.0.2
```
Comment (gitlawr) [inline deploy/charts/harvester/Chart.lock]: I don't find longhorn in the subcharts

File: deploy/charts/harvester/dependency_charts/multus/values.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  # remove it if your masters can't run pods
+  - key: node-role.kubernetes.io/master
+    effect: NoSchedule
+
+affinity: {}
+
+## Specify the cluster type, default to k3s by the harvester
+## with k3s cluster, the minimum version with v1.19.4+k3s1 is required
+cluster:
+  type: k3s
```
Comment (gitlawr) [inline deploy/charts/harvester/dependency_charts/multus/values.yaml]: I think we should make cni-bin-dir & cni-conf_dir configurable. Cluster type is obscure/unclear

File: deploy/charts/harvester/Chart.lock (hunk 1 above)
Comment (guangbochen) [reply deploy/charts/harvester/Chart.lock]: removed

File: deploy/charts/harvester/dependency_charts/multus/values.yaml (hunk 2 above)
Comment (guangbochen) [reply deploy/charts/harvester/dependency_charts/multus/values.yaml]: updated

File: deploy/charts/harvester/dependency_charts/multus/templates/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    spec:
+      hostNetwork: true
+      serviceAccountName: {{ include "multus.name" . }}
+      containers:
+        - name: kube-multus
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: ["/entrypoint.sh"]
+          args:
+            - "--multus-conf-file=auto"
```
Comment (gitlawr) [inline deploy/charts/harvester/dependency_charts/multus/templates/daemonset.yaml:27]: ~~I think we also need to set `--multus-kubeconfig-file-host` here, can you confirm?~~ NVM

File: deploy/charts/harvester/dependency_charts/multus/values.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+affinity: {}
+
+## Specify multus-cni daemonSet extra arguments
+args: []
+
+## Specify the cni host path
+hostPath:
+  cni: /etc/cni/net.d
+  cnibin: /opt/cni/bin
```
[the remark this answers is not in the document]
Comment (guangbochen) [reply deploy/charts/harvester/dependency_charts/multus/values.yaml]: done

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**